docs/examples/word_embedding/word_embedding.ipynb

```diff
@@ -653,16 +653,16 @@
    "outputs": [],
    "source": [
     "def norm_vecs_by_row(x):\n",
-    "    return x / nd.sqrt(nd.sum(x * x, axis=1)).reshape((-1,1))\n",
+    "    return x / nd.sqrt(nd.sum(x * x, axis=1) + 1E-10).reshape((-1,1))\n",
     "\n",
     "def get_knn(vocab, k, word):\n",
     "    word_vec = vocab.embedding[word].reshape((-1, 1))\n",
     "    vocab_vecs = norm_vecs_by_row(vocab.embedding.idx_to_vec)\n",
     "    dot_prod = nd.dot(vocab_vecs, word_vec)\n",
-    "    indices = nd.topk(dot_prod.reshape((len(vocab), )), k=k+5, ret_typ='indices')\n",
```

Comment:
The reason why special tokens were retrieved is that their word vectors were `nan` and the `topk` operator considered `nan` as the highest ranking. With the changes here the special tokens will not be among the topk elements, consequently instead of k + 5 we need k + 1 to only exclude the input token.

Reply:
With mxnet 1.3 the `topk` operator however gets confused by the `nan` values and returns some 'random' words instead of the topk words if `nan` is present in the input.

Reply:
Yes, they are initialized as 0 and consequently there was a division by 0 in `norm_vecs_by_row`